src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -168,6 +198,24 @@ public static void Run(ListSourceArgs args, Func<ILogger> getLogger)
                             }
                             legend += " ";
                             getLogger().LogMinimal(legend + source.Source);
+
+                            if (source.IsHttp && !source.IsHttps)
+                            {
+                                if (httpPackageSources == null)
+                                {
+                                    httpPackageSources = new();
+                                }
+                                httpPackageSources.Add(source);
```

Comment:
If only single source is expected, then why bother creating `httpPackageSources` list? it's unnecessary allocation.
Probably just do same thing you did in `src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs` file.

Reply:
This is the list command so there are multiple sources, here I'm iterating through all the sources and storing the HTTP sources in a new list so I can print them later